src/Views/Bookmark.elm

```diff
@@ -291,25 +294,68 @@ bookmarkView cfg ({ name, query, version } as bookmark) =
                 Bookmark.Veli veliQuery ->
                     Just veliQuery
                         |> Route.ObjectSimulator Scope.Veli cfg.impact.trigram
+
+        beingRenamed =
+            case cfg.bookmarkBeingRenamed of
+                Just renamedBookmark ->
+                    renamedBookmark.query == bookmark.query
+
+                _ ->
+                    False
+
+        renameButton bk =
+            button
+                [ type_ "button"
+                , class "btn btn-sm btn-info"
+                , title "Renommer"
+                , attribute "aria-label" "Renommer"
+                , onClick (cfg.updateRenamedBookmarkName bk bk.name)
+                ]
+                [ Icon.pencil ]
     in
     li
         [ class "list-group-item d-flex justify-content-between align-items-center gap-1 fs-7"
         , classList [ ( "active", query == currentQuery ) ]
         ]
         [ VersionView.view version
-        , a
-            [ class "flex-fill text-truncate"
-            , classList [ ( "active text-white", query == currentQuery ) ]
-            , bookmark
-                |> Bookmark.toQueryDescription cfg.session.db
-                |> title
-            , bookmarkRoute
-                |> Route.toString
-                |> (++) cfg.session.clientUrl
-                |> href
-            ]
-            [ text name
-            ]
+        , case ( beingRenamed, cfg.bookmarkBeingRenamed ) of
+            ( True, Just renamedBookmark ) ->
+                input
+                    [ type_ "text"
+                    , class "form-control"
```

Comment:
I suggest adding the `form-control-sm` to avoid the line to flicker when switching from label to text input